File: Plugins/Aspose_Email_Java_for_Ruby/lib/asposeemailjava/Outlook/addmapitasktopst.rb
module Asposeemailjava
  module AddMapiTaskToPST
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        task = Rjb::import('com.aspose.email.MapiTask').new("To Do", "Just click and type to add new task", Rjb::import('java.util.Date').new, Rjb::import('java.util.Date').new)
        task.setPercentComplete(20)
        task.setEstimatedEffort(2000)
        task.setActualEffort(20)
        task.setHistory(Rjb::import('com.aspose.email.MapiTaskHistory').Assigned)
        task.setLastUpdate(Rjb::import('java.util.Date').new)
        task.getUsers().setOwner("Darius")
        task.getUsers().setLastAssigner("Harkness")
        task.getUsers().setLastDelegate("Harkness")
        task.getUsers().setOwnership(Rjb::import('com.aspose.email.MapiTaskOwnership').AssignersCopy)

        pst = Rjb::import('com.aspose.email.PersonalStorage').create(data_dir + "TaskPST.pst", Rjb::import('com.aspose.email.FileFormatVersion').Unicode)
        task_folder = pst.createPredefinedFolder("Tasks", Rjb::import('com.aspose.email.StandardIpmFolder').Tasks)
        task_folder.addMapiMessageItem(task)

        puts "Added MapiTask Successfully."
    end
  end
end

File: Examples/src/main/java/com/aspose/email/examples/outlook/pst/AddMapiTaskToPST.java
package com.aspose.email.examples.outlook.pst;

import java.util.Date;

import com.aspose.email.FileFormatVersion;
import com.aspose.email.FolderInfo;
import com.aspose.email.MapiTask;
import com.aspose.email.MapiTaskHistory;
import com.aspose.email.MapiTaskOwnership;
import com.aspose.email.PersonalStorage;
import com.aspose.email.StandardIpmFolder;
import com.aspose.email.examples.Utils;

public class AddMapiTaskToPST {
	
	public static String dataDir = Utils.getSharedDataDir(AddMapiTaskToPST.class) + "outlook/";
	
	public static void main(String[] args) {
		MapiTask task = new MapiTask("To Do", "Just click and type to add new task", new Date(), new Date());
		task.setPercentComplete(20);
		task.setEstimatedEffort(2000);
		task.setActualEffort(20);
		task.setHistory(MapiTaskHistory.Assigned);
		task.setLastUpdate(new Date());
		task.getUsers().setOwner("Darius");
		task.getUsers().setLastAssigner("Harkness");
		task.getUsers().setLastDelegate("Harkness");
		task.getUsers().setOwnership(MapiTaskOwnership.AssignersCopy);

		PersonalStorage pst = PersonalStorage.create(dataDir + "TaskPST_out.pst", FileFormatVersion.Unicode);
		FolderInfo taskFolder = pst.createPredefinedFolder("Tasks", StandardIpmFolder.Tasks);
		taskFolder.addMapiMessageItem(task);
	}

}
